app/Material.php

```diff
@@ -47,4 +49,41 @@ public function isLost()
     {
         return !is_null($this->returner_registration_mark);
     }
+
+    /**
+     * Scope a query to only return borrowable materials
+     *
+     * @param Builder $query
+     * @return Builder $query
+     */
+    public function scopeBorrowable($query)
+    {
+        return $query->where([
+            ['amount', '>=', 1],
+            ['returner_registration_mark', '=', null],
+            ['tooker_registration_mark', '=', null]
+        ]);
+    }
+
+    public function hasBorrowableWithName(string $name)
+    {
+        $material = $this->borrowable()->where('name', $name)->first();
+        return isset($material);
+    }
+
+    public function isAnBorrowableAmount(int $materialAmount)
+    {
+        return (! $this->amount) || ($this->amount < $materialAmount);
```

Comment:
Amount have as default value 1, in the other words it always will be TRUE.

Reply:
That doesn't make sense, I could loan a material and it would have the amount 0.

Reply:
But that logic have semantic flaws, in the method name I said isAnBorrowableAmount, but in the logic I went right to the isn't. I refactored it now.